src/librustc_typeck/check/writeback.rs

```diff
@@ -480,25 +481,33 @@ impl<'cx, 'gcx, 'tcx> WritebackCx<'cx, 'gcx, 'tcx> {
                                         span,
                                         format!(
                                             "lifetime `{}` is part of concrete type but not used \
-                                            in parameter list of existential type",
+                                             in parameter list of existential type",
                                             region,
                                         ),
                                     )
                                     .emit();
                                 self.tcx().global_tcx().mk_region(ty::ReStatic)
                             }
                         }
-                    }
+                    },
                 })
             };
 
-            let old = self.tables.concrete_existential_types.insert(def_id, definition_ty);
+            if let ty::TyAnon(def_id, _substs) = definition_ty.sty {
```

Comment:
I think we should compare this `def_id` to the other `def_id` in scope and only run the `type_of` query if it is the same. Otherwise we're running it uselessly every time an existential type resolves to an existential type or an impl Trait type